### ISSUE TITLE
Asynchronous initialization causing concurrently problem in DashboardApplication

### ISSUE THREAD

**Issue by jasonjoo2010** [alibaba/Sentinel#705]
## Issue Description
Some racing problem occurred like ConcurrentModification on System.properties

### Describe what happened (or what feature you want)
![E9B3A872-4232-406d-8976-FE114B2DA775](https://user-images.githubusercontent.com/17573073/56719741-f15fd400-6773-11e9-9161-e1d47aed9d93.png)

### How to reproduce it (as minimally and precisely as possible)
Calling setProperty putting new keys into property set while InitExecutor.doInit is executed asynchronously.

### Asynchronism Supporting
We should declare whether asynchronism should be supported here.  
We take the answer `YES` till someone submitted different opinions.

### Direct Cause of the Issue
The main method is shown below used by `Dashboard` booting up.

```java
public class DashboardApplication {

    public static void main(String[] args) {
        triggerSentinelInit();
        SpringApplication.run(DashboardApplication.class, args);
    }

    private static void triggerSentinelInit() {
        new Thread(() -> InitExecutor.doInit()).start();
    }
}
```

The `InitExecutor.doInit()` is called in another thread when spring boot is starting.
It may add new keys calling `System.setProperty` when booting. And we have the code below to load config from properties in `SentinelConfig.loadProps`:

```java
        // JVM parameter override file config.
        for (Map.Entry<Object, Object> entry : System.getProperties().entrySet()) {
            String configKey = entry.getKey().toString();
            String configValue = entry.getValue().toString();
            String configValueOld = getConfig(configKey);
            SentinelConfig.setConfig(configKey, configValue);
            if (configValueOld != null) {
                RecordLog.info("[SentinelConfig] JVM parameter overrides {0}: {1} -> {2}", configKey, configValueOld, configValue);
            }
        }
```

Which we may use CopyOnWriteArraySet instead to walk through the values safely.

Is there any opinions, folks?

**Comment by sczyh30** [comment]
Yes, it's necessary to make it thread-safe.

**Comment by sczyh30** [comment]
Fixed in #706
